extensions/amp-minute-media-player/0.1/amp-minute-media-player.js

```diff
@@ -0,0 +1,480 @@
+      );
+    }, 3000);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    //Define which layouts our element support
+
+    //Size-defined layouts: fixed, fixed height, responsive and fill
+    return isLayoutSizeDefined(layout);
+    //return layout == Layout.RESPONSIVE;
+  }
+
+  /**
+   * Init Semantic params
+   * @private
+   */
+  initSemanticFields_() {
+    const {element} = this;
+    this.scannedElement_ = element.getAttribute('data-scanned-element') || '';
+    this.tags_ = element.getAttribute('data-tags') || '';
+    this.minimumDateFactor_ =
+      element.getAttribute('data-minimum-date-factor') || '';
+    //console.log(element.getAttribute('data-scanned-element-type'));
+    this.scannedElementType_ =
+      element.getAttribute('data-scanned-element-type') || '';
+  }
+
+  /**
+   * @param {!Event} event
+   * @private
+   */
+  handleMPlayerMessage_(event) {
+    if (
+      !originMatches(
+        event,
+        this.iframe_,
+        'https://syringe.s3-us-west-2.amazonaws.com'
+      )
+    ) {
+      return;
+    }
+    const eventData = getData(event);
+    if (!isJsonOrObj(eventData)) {
+      return;
+    }
+    const data = objOrParseJson(eventData);
+    //console.log(data); DONT GET HERE
+    if (data === undefined) {
+      return; // We only process valid JSON.
+    }
+
+    const eventType = data['event'];
+    //console.log(eventType);
+    if (eventType == 'ready') {
+      /*** NECESSARY?? ***/
+      Services.timerFor(this.win).cancel(this.readyTimeout_);
+    }
+    redispatch(this.element, eventType, {
+      'playing': VideoEvents.PLAYING,
+      'paused': VideoEvents.PAUSE,
+      'muted': VideoEvents.MUTED,
+      'unmuted': VideoEvents.UNMUTED,
+      'ended': [VideoEvents.ENDED, VideoEvents.PAUSE],
+      'ads-ad-started': VideoEvents.AD_START,
+      'ads-ad-ended': VideoEvents.AD_END,
+    });
+  }
+
+  /**
+   * Build Iframe source
+   * @return {string}
+   * @private
+   */
+  iframeSource_() {
+    const {element} = this;
+    const source =
+      'https://syringe.s3-us-west-2.amazonaws.com/dev/amp/mplayer.html' +
+      `?content_type=${encodeURIComponent(this.contentType_)}` +
+      (this.contentId_ !== ''
+        ? `&content_id=${encodeURIComponent(this.contentId_)}`
+        : (this.scannedElement_ != ''
+            ? '&scanned_element=' +
+              `${encodeURIComponent(this.scannedElement_)}`
+            : '') +
+          (this.tags_ != '' ? `&tags=${encodeURIComponent(this.tags_)}` : '') +
+          (this.minimumDateFactor_ != ''
+            ? '&minimum_date_factor=' +
+              `${encodeURIComponent(this.minimumDateFactor_)}`
+            : '') +
+          (this.scannedElementType_
+            ? '&scanned_element_type=' +
+              `${encodeURIComponent(this.scannedElementType_)}`
+            : ''));
+
+    const moreQueryParams = dict({
+      'player_id': element.getAttribute('data-player-id') || undefined,
```

Comment:
If you're already using `addParamsToUrl`, construct the previous string by modifying the `dict()`. This string concatenation scheme is very unreadable as it stands.

Reply:
Changed it. 
Thanks